src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -726,11 +726,14 @@ private async Task<List<LaunchCommand>> GetInitializeCommands()
 
                     // check for remote
                     string destination = localLaunchOptions?.MIDebuggerServerAddress;
+                    bool useExtendedRemote = localLaunchOptions?.UseExtendedRemote ?? false;
                     if (!string.IsNullOrWhiteSpace(destination))
                     {
-                        commands.Add(new LaunchCommand("-target-select remote " + destination, string.Format(CultureInfo.CurrentCulture, ResourceStrings.ConnectingMessage, destination)));
+                        string remoteMode = useExtendedRemote ? "extended-remote" : "remote";
+                        commands.Add(new LaunchCommand($"-target-select {remoteMode} {destination}", string.Format(CultureInfo.CurrentCulture, ResourceStrings.ConnectingMessage, destination)));
                     }
-                    else // gdbserver is already attached when using LocalLaunchOptions
+                    // Allow attach after connection only in extended-remote mode
+                    if (useExtendedRemote || (!useExtendedRemote && string.IsNullOrWhiteSpace(destination)))
```

Comment:
this block falls into adding a -target-attach command. what happens if gdbserver is launched with --attach and then we issue a -target-attach command via extended-remote connection?

Reply:
This is the -target-attach command. There is no additional --attach even under attach mode.

Reply:
Ah sorry I misread your question. attach will fail with a message like `failed to attach to process....`